src/Halogen/Infinite/Scroll.purs

```diff
@@ -186,14 +196,20 @@ handleFeedAction (PageOutput (PageHeight {id, height})) = do
                           }
     setHeight Nothing = Nothing
     setHeight (Just page) = Just (page { pageHeight = height })
-handleFeedAction (PageOutput (PageIntersection {id, height})) = do
-  H.modify_ setPageIntersection
-  H.getRef (H.RefLabel "feed") >>= traverse_ manageFeed
-  where
-    setPageIntersection st = st { pages = Map.alter setIntersection id st.pages
-                                }
-    setIntersection Nothing = Nothing
-    setIntersection (Just page) = Just (page { visibleHeight = height })
+handleFeedAction (PageOutput (PageIntersection page)) = mask do
+  feed' <- H.getRef (H.RefLabel "feed")
+  flip traverse_ feed' $ \feed -> do  
+    st <- H.modify (\st -> execState (stateUpdate (computeUpdate st page)) st) 
+    t <- H.liftEffect $ scrollTop feed
+    H.liftEffect $ traverse_ (\o -> setScrollTop (t + o) feed) st.update.scroll
```

Comment:
This still doesn't play nicely with all browsers when loading/unloading at the top.